src/main/scala/com/monovore/decline/Argument.scala

```diff
@@ -5,7 +5,7 @@ import java.nio.file.{InvalidPathException, Path, Paths}
 
 import cats.data.{Validated, ValidatedNel}
 
-trait Argument[A] {
+trait Argument[A] { self =>
```

Comment:
we could seal Argument and give a constructor function to create it. I kind of like that so it discourages people from doing OO designs and putting these things in funny places.

Also, you can add some combinator methods on Argument, which I did, then deleted.